taglib/fileref.cpp

```diff
@@ -284,3 +289,77 @@ File *FileRef::create(FileName fileName, bool readAudioProperties,
 
   return 0;
 }
+
+File *FileRef::create(IOStream* ioStream, bool readAudioProperties, AudioProperties::ReadStyle audioPropertiesStyle)
+{
+    // Ok, this is really dumb for now, but it works for testing.
+
+    String ext;
+    {
+  #ifdef _WIN32
+
+      String s = ioStream->name().toString();
+
+  #else
+
+      String s = ioStream->name();
+
+   #endif
+
+      const int pos = s.rfind(".");
+      if(pos != -1)
+        ext = s.substr(pos + 1).upper();
+    }
+
+    // If this list is updated, the method defaultFileExtensions() should also be
+    // updated.  However at some point that list should be created at the same time
+    // that a default file type resolver is created.
+
+    if(!ext.isEmpty()) {
```

Comment:
Please do not make a copy of this whole block. Create an internal template function instead.